eppoclient/client.go

```diff
@@ -82,25 +95,30 @@ func (ec *EppoClient) GetJSONAssignment(flagKey string, subjectKey string, subje
 
 func (ec *EppoClient) getAssignment(flagKey string, subjectKey string, subjectAttributes Attributes, variationType variationType) (interface{}, error) {
 	if subjectKey == "" {
+		ec.applicationLogger.Error("no subject key provided")
 		panic("no subject key provided")
 	}
 
 	if flagKey == "" {
+		ec.applicationLogger.Error("no flag key provided")
 		panic("no flag key provided")
 	}
 
 	flag, err := ec.configRequestor.GetConfiguration(flagKey)
 	if err != nil {
+		ec.applicationLogger.Info("failed to get flag configuration: %v", err)
 		return nil, err
 	}
 
 	err = flag.verifyType(variationType)
 	if err != nil {
+		ec.applicationLogger.Info("failed to verify flag type: %v", err)
 		return nil, err
 	}
 
-	assignmentValue, assignmentEvent, err := flag.eval(subjectKey, subjectAttributes)
+	assignmentValue, assignmentEvent, err := flag.eval(subjectKey, subjectAttributes, ec.applicationLogger)
```

Comment:
passing the logger around isn't my favorite pattern; suggestions welcome

Reply:
@rasendubi I have seen the `context` used as a way to pass around things like application logger, metrics store, etc. However this is kind of an implementation detail and we can iterate on improving it.